src/buzz/buzzmath.c

```diff
@@ -159,7 +159,7 @@ int buzzmath_floor(buzzvm_t vm) {
    /* Get argument */
    buzzvm_lload(vm, 1);
    buzzobj_t o = buzzvm_stack_at(vm, 1);
-   if(o->o.type == BUZZTYPE_FLOAT)    buzzvm_pushf(vm, floor(o->f.value));
+   if(o->o.type == BUZZTYPE_FLOAT)    buzzvm_pushi(vm, floor(o->f.value));
```

Comment:
This line would fail if the type passed to floor is integer. This would make it difficult to program with this Buzz function. Please add a type check that says that if the type is integer, you return the exact same value.